libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -144,14 +140,28 @@ bool DeclarationContainer::registerDeclaration(
 	return registerDeclaration(_declaration, nullptr, nullptr, _invisible, _update);
 }
 
-vector<Declaration const*> DeclarationContainer::resolveName(ASTString const& _name, bool _recursive, bool _alsoInvisible) const
+vector<Declaration const*> DeclarationContainer::resolveName(
+	ASTString const& _name,
+	bool _recursive,
+	bool _alsoInvisible,
+	bool _onlyVisibleAsUnqualifiedNames
+) const
 {
 	solAssert(!_name.empty(), "Attempt to resolve empty name.");
 	vector<Declaration const*> result;
+
+	auto filter = [&result, &_onlyVisibleAsUnqualifiedNames](vector<Declaration const*> _declarations) {
+		_onlyVisibleAsUnqualifiedNames ?
+			copy_if(_declarations.cbegin(), _declarations.cend(), back_inserter(result), [](auto& _declaration) {
+				return _declaration->isVisibleAsUnqualifiedName();
+			}) :
+			copy(_declarations.cbegin(), _declarations.cend(), back_inserter(result));
+	};
```

Comment:
Maybe this would be a bit nicer with ranges-v3.

Reply:
Done. Unfortunately ranges-v3 version is not as efficient as this (requires an extra copy) because we do not have `+=` overloaded for ranges and the range needs to be materialized.